test/Microsoft.Framework.DependencyInjection.Tests/ActivatorUtilitiesTests.cs

```diff
@@ -220,5 +220,17 @@ public void GetServiceOrCreateInstanceUnregisteredService()
             Assert.Equal(2, service.InstanceId);
             Assert.Equal(2, CreationCountFakeService.InstanceCount);
         }
+
+        [Theory]
+        [MemberData(nameof(CreateInstanceFuncs))]
+        public void UnRegisteredServiceAsConstructorParameterThrowsException(CreateInstanceFunc createFunc)
+        {
+            var serviceProvider = new ServiceCollection()
+                .AddSingleton<CreationCountFakeService>()
+                .BuildServiceProvider();
+
+            Assert.Throws<InvalidOperationException>(() =>
```

Comment:
We should assert that we get the expected exception message like we do in the other tests that use Assert.Throws.

Right now, the exception thrown CreateInstance has a different message than the one thrown from CreateFactory. The exception message from CreateInstance is a little better since it tells what type is being activated.

This will mean that you can no longer use GetRequiredService:

```C#
    private static object GetService(IServiceProvider sp, Type type, Type requiredBy, bool serviceRequired)
    {
        var service = sp.GetService(type);

        if (service == null && serviceRequired)
        {
            throw new InvalidOperationException(
                Resources.FormatCannotResolveService(type, requiredBy));
        }

        return service;
    }
```

I noticed while writing this comment that CreateInstance doesn't provide the format parameters for FormatCannotResolveService in the correct order, so you could fix that up too as part of this change.